data/list/basic.lean

```diff
@@ -1160,6 +1160,22 @@ by induction l; simp [list.join, *] at *
 
 end monoid
 
+@[simp, to_additive list.sum_erase]
+theorem prod_erase [decidable_eq α] [comm_monoid α] {a} : 
+  Π {l : list α}, a ∈ l → a * (l.erase a).prod = l.prod
+| []          := λ h, absurd h $ not_mem_nil _
```

Comment:
the proof is easier by following the proof of `a ∈ l` instead of following the construction of `l`

Reply:
sorry, now I see why it doesn't work, the second branch of `a ∈ l` doesn't tell us with `b::l` `b ~= a`

Reply:
Yeah I have tried that before but later found that I have to do a by_cases anyway. Thanks for the comments!